src/mpid/ch4/netmod/ofi/ofi_win.h

```diff
@@ -1062,6 +1068,8 @@ static inline int MPIDI_NM_mpi_win_allocate_shared(MPI_Aint size,
         shm_unlink(shm_key);
 
   fn_exit:
+    if (shm_key != NULL)
```

Comment:
The problem without the check is that in some places before the `(re)alloc` there is a `goto fn_exit` or `goto fn_fail` (which in turn jumps `goto fn_exit`). This means that when we reach fn_exit, it is possible that the `(re)alloc` has not been called yet :(